test/unit/federation_server_test.js

```diff
@@ -227,5 +227,98 @@ FEDERATION_SERVER="https://api.stellar.org/federation"
           .then(() => tempServer.close());
       });
     });
+
+  });
+
+  describe('FederationServer times out when response lags and timeout set', function () {
+    it("resolveAddress times out", function (done) {
+      StellarSdk.Config.setTimeout(100);
+
+      // Unable to create temp server in a browser
+      if (typeof window != 'undefined') {
+        return done();
+      }
+
+      let tempServer = http.createServer((req, res) => {
+        setTimeout(() => {}, 10000);
+      }).listen(4444, () => {
+        new StellarSdk.FederationServer('http://localhost:4444/federation', 'stellar.org', {allowHttp: true})
+          .resolveAddress('bob*stellar.org')
+          .should.be.rejectedWith(/timeout of 100ms exceeded/)
+          .notify(done)
+          .then(() => tempServer.close());
+      });
+    });
+
+    it("resolveAccountId times out", function (done) {
+      StellarSdk.Config.setTimeout(100);
+
+      // Unable to create temp server in a browser
+      if (typeof window != 'undefined') {
+        return done();
+      }
+
+      let tempServer = http.createServer((req, res) => {
+        setTimeout(() => {}, 10000);
+      }).listen(4444, () => {
+        new StellarSdk.FederationServer('http://localhost:4444/federation', 'stellar.org', {allowHttp: true})
+          .resolveAccountId('GB5XVAABEQMY63WTHDQ5RXADGYF345VWMNPTN2GFUDZT57D57ZQTJ7PS')
+          .should.be.rejectedWith(/timeout of 100ms exceeded/)
+          .notify(done)
+          .then(() => tempServer.close());
+      });
+    });
+
+    it("resolveTransactionId times out", function (done) {
+      StellarSdk.Config.setTimeout(100);
+
+      // Unable to create temp server in a browser
+      if (typeof window != 'undefined') {
+        return done();
+      }
+
+      let tempServer = http.createServer((req, res) => {
+        setTimeout(() => {}, 10000);
+      }).listen(4444, () => {
+        new StellarSdk.FederationServer('http://localhost:4444/federation', 'stellar.org', {allowHttp: true})
```

Comment:
The last missing test case is creating a new `FederationServer` with `opts.timeout` set (without calling `Config.setTimeout`).

I think you can create a loop that will test all the methods (`resolveTransactionId`, `resolveAccountId`, ...) with both cases: `Config` in the first iteration and `opts` in the second. Without it we will have a lot of code duplication here.